sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/ClientSideRequestStatistics.java

```diff
@@ -210,6 +265,27 @@ void setRegionsContacted(Set<URI> regionsContacted) {
         this.regionsContacted = regionsContacted;
     }
 
+    private void printSystemInformation(StringBuilder stringBuilder) {
+        try {
+            long totalMemory = Runtime.getRuntime().totalMemory() / 1024;
+            long freeMemory = Runtime.getRuntime().freeMemory() / 1024;
+            long maxMemory = Runtime.getRuntime().maxMemory() / 1024;
+            String usedMemory = totalMemory - freeMemory + " KB";
+            String availableMemory = (maxMemory - (totalMemory - freeMemory)) + " KB";
+
+            String processCpuLoad = Double.toString(this.mbean.getProcessCpuLoad());
+            String systemCpuLoad = Double.toString(this.mbean.getSystemCpuLoad());
+
+            stringBuilder.append("System State Information -------").append(System.lineSeparator());
+            stringBuilder.append("Used Memory : " + usedMemory).append(System.lineSeparator());
+            stringBuilder.append("Available Memory : " + availableMemory).append(System.lineSeparator());
+            stringBuilder.append("CPU Process Load : " + processCpuLoad).append(System.lineSeparator());
+            stringBuilder.append("CPU System Load : " + systemCpuLoad).append(System.lineSeparator());
+        } catch (Exception e) {
```

Comment:
can anything in the try/catch throw?

Reply:
Intention is be safe and not affect the other diagnostics, please suggest if you think otherwise

Reply:
it is good to be safe. looks good. I was just curious if you have seen any specific exception getting thrown from here.

in general diagnostics code should be written in a way that never throws. you did the right thing. thank you